src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -173,7 +173,9 @@ type ErrorCode =
     | ResultComparisonNotInOperationIf = 5024
     | ReturnInResultConditionedBlock = 5025
     | SetInResultConditionedBlock = 5026
-    | UnsupportedCapability = 5027
+    | UnsupportedCallableCapability = 5027
+    // TODO: RELEASE 2021-07: Remove ErrorCode.UnsupportedCapability.
+    | [<Obsolete "Renamed to UnsupportedCallableCapability.">] UnsupportedCapability = 5027
```

Comment:
What is the reason for renaming it? In general, there is always a temptation to rename items, but since it is a burden to the user (even with a nice deprecation), we should try to avoid unnecessarily deprecating/renaming things.

Reply:
The previous name is misleading, and it confused me several times while working on this PR. `UnsupportedCapability` makes it sound like it's generated for every capability error, but it's only used when an unsupported callable is referenced.

Reply:
Ok.